File: src/main/java/studio/karllang/karl/parser/ast/values/BooleanValue.java
package studio.karllang.karl.parser.ast.values;

import studio.karllang.karl.parser.TokenType;

public class BooleanValue extends Value {
    private final Boolean value;

    public BooleanValue(Boolean value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return value.toString();
    }

    @Override
    public int toInt() {
        return 0;
    }

    @Override
    public float toFloat() {
        return 0;
    }

    @Override
    public TokenType getType() {
        return TokenType.BOOL_VALUE;
    }
}

File: src/main/kotlin/studio/karllang/karl/parser/ast/values/BooleanValue.kt
package studio.karllang.karl.parser.ast.values

import studio.karllang.karl.lexer.TokenType

class BooleanValue(private val value: Boolean) : Value() {
    override val type: TokenType = TokenType.BOOL_VALUE

    override fun toString(): String {
        return value.toString()
    }

    override fun toInt(): Int {
        return 0
    }

    override fun toFloat(): Float {
        return 0f
    }
}